integration-testing/test/cl_node/casperlabs_network.py

```diff
@@ -181,21 +194,32 @@ def create_cl_network(self):
 
 class MultiNodeJoinedNetwork(CasperLabsNetwork):
 
-    def create_cl_network(self, node_count=10):
+    def __init__(self, n: int, docker_client: 'DockerClient', extra_docker_params = {}):
+        self._n = n
+        self.extra_docker_params = extra_docker_params
+        super(MultiNodeJoinedNetwork, self).__init__(docker_client)
+
+
+    def docker_config(self):
         kp = self.get_key()
         config = DockerConfig(self.docker_client,
                               node_private_key=kp.private_key,
-                              node_public_key=kp.public_key,
-                              network=self.create_docker_network())
+                              node_public_key=kp.public_key,)
+
+        return dataclasses.replace(config, **self.extra_docker_params)
+        
+
+    def create_cl_network(self):
+        config = self.docker_config()
+        config.network = self.create_docker_network()
         self.add_bootstrap(config)
 
-        for _ in range(1, node_count):
-            kp = self.get_key()
-            config = DockerConfig(self.docker_client, node_private_key=kp.private_key)
-            self.add_cl_node(config)
+        for _ in range(1, self._n):
+            self.add_cl_node(self.docker_config())
```

Comment:
```suggestion
            self.add_cl_node(config)
```
Not sure what the intention is, but `self.docker_config()` creates a new instance each time, so `config.network` is not going to be there on subsequent ones passed to `add_cl_node`.

Reply:
Good catch, thanks. Calling the method again was not needed.

Reply:
Actually, it turns out we can't reuse the `config` object. Each time `docker_config` is called it generates a new, unique key.

Reply:
Okay but does only the first one need the network?

Reply:
So, the first one is the bootstrap. We set `network` for it. Then in `add_cl_node` we reuse `network` of the bootstrap (`self.cl_nodes[0].node.network`) for other nodes.

Reply:
Right, I didn't see that code before, thanks.